torch/nn/parallel/distributed.py

```diff
@@ -223,21 +223,26 @@ def __init__(self, module, device_ids=None,
         )
 
         self.is_multi_device_module = len({p.device for p in module.parameters()}) > 1
-        self.is_cuda = all([p.device.type == 'cuda' for p in module.parameters()])
-
-        if not self.is_cuda or self.is_multi_device_module:
+        distinct_device_type = {p.device.type for p in module.parameters()}
```

Comment:
rename this var to `distinct_device_types` as it is a set and it could contain multiple device types?